packages/minter-contracts/ligo/src/swaps/fa2_allowlisted_swap_offchain_nocounter.mligo

```diff
@@ -0,0 +1,46 @@
+#include "fa2_allowlisted_swap.mligo"
+#include "../common.mligo"
+
+type permit_accept_param = 
+  [@layout:comb]
+  {
+    swap_id : nat;
+    permit : permit;
+  }
+
+type offchain_swap_entry_points =
+  | BaseSwap of entrypoints
+  | Offchain_accept of permit_accept_param list
+
+let offchain_accept(p, storage, ops : permit_accept_param * storage * operation list) 
+  : allowlist_return = begin
+    fail_if_not_admin(storage.admin);
+    let param_hash = Crypto.blake2b (Bytes.pack p.swap_id) in 
+    check_permit (p.permit, 0n, param_hash);
+    let swap_accepter = address_from_key (p.permit.signerKey) in
+    let swap = get_swap(p.swap_id, storage.swap) in
+    let ops = accept_swap_update_ops_list(swap, swap_accepter, ops, storage.swap) in
+    let swap_storage = accept_swap_update_storage(p.swap_id, swap, swap_accepter, storage.swap) in
+    (ops, {storage with swap = swap_storage})
+  end
+
+let rec offchain_accept_batch_helper(ps, (ops, storage) : permit_accept_param list * (operation list * storage)) 
+  : allowlist_return = 
+    match ps with 
+      | [] -> (ops, storage)
+      | permit :: remaining_permits -> 
```

Comment:
would it be straightforward to replace this with a fold over the list? it gets compiled to a `LOOP_LEFT` with `IF_UNCONS`, etc

Reply:
Fixed!